id/mod.go

```diff
@@ -1,7 +1,60 @@
 package id
 
+import (
+	"crypto/sha256"
+	"go.dedis.ch/dela/mino"
+	"math/big"
+)
+
 type Id interface {
-	GetLength() byte
+	GetLength() int
 	GetBase() byte
 	GetDigit(pos int) byte
-}
\ No newline at end of file
+}
+
+type ArrayId struct {
+	Id   []byte
+	Base byte
+}
+
+// Constructs an Id by taking a hash of the address modulo (base ^ len), then presenting the id as an array of digits.
+func MakeArrayId(address mino.Address, base byte, len int) Id {
```

Comment:
A more idomatic name would be `NewArrayNodeID` with a return type of `ArrayNodeID`. Please ensure the comment wraps around 80 chars.

Would be useful to have a unit test for this